rfcs/040-tei_adapter/README.md

```diff
@@ -68,43 +68,117 @@ The biggest disadvantage is that we don't have control or visibility if the Webh
+    "url": "https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/git/blobs/ed307046335248d38c18cb8320d6ca9253e0d35b"
+  },
+  {
+    "path": ".gitattributes",
+    "mode": "100644",
+    "type": "blob",
+    "sha": "bcb42237cafe6a22e80b9d720dcbf6395198d249",
+    "size": 62,
+    "url": "https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/git/blobs/bcb42237cafe6a22e80b9d720dcbf6395198d249"
+  },
+  {
+    "path": "Arabic",
+    "mode": "040000",
+    "type": "tree",
+    "sha": "187c8c750739ee6a8f9f99ebf7420a46457670d5",
+    "url": "https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/git/trees/187c8c750739ee6a8f9f99ebf7420a46457670d5"
+  },
+  {
+    "path": "Arabic/README.md",
+    "mode": "100644",
+    "type": "blob",
+    "sha": "4bfe74311d86293447f173108190a4b4664d68ea",
+    "size": 1753,
+    "url": "https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/git/blobs/4bfe74311d86293447f173108190a4b4664d68ea"
+  },
+  {
+    "path": "Arabic/WMS_Arabic_1.xml",
+    "mode": "100644",
+    "type": "blob",
+    "sha": "17f54c8c5ebccec643e23b54b3f4277f86204282",
+    "size": 59802,
+    "url": "https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/git/blobs/17f54c8c5ebccec643e23b54b3f4277f86204282"
+  },
+...
+  ]
+}
+```
+Every entry in the `tree` can be either a `tree` or a `blob`. A `blob` is a file in the repo and has a hash associated to it. 
+If the content of a file changes, the hash changes. 
+This means that we can query for the tree in master at regular intervals and store the last version of the tree in S3. 
+At every request, we can compare the old tree with the new one to know what changes have happened.
 ### TEI Adapter
 ![](tei_adapter_stack.png)
-The proposal is to make the TEI adapter 2 services
-#### GitHub Api service
 
-It receives time windows from the lambda and for each one:
+The proposal is to have one Lambda that queries the GitHub API and a TEI adapter service that retrieves the files that 
+have been modified and stores them in a VHS.
+#### Tree updater Lambda 
+
+This is a scheduled Lambda that:
   
-- It calls the GitHub API to get the commits within that window. 
-  This is an example of a call to GitHub API and returns a list of commits within the time window passed in the query parameters:
-``` 
-GET https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/commits?since=2021-01-20T10:00:00&until=2021-03-22T18:01:00&branch=master
+- calls the GitHub API to get the latest tree.
+- Retrieves the old tree from S3.
+- Compares the two and for every change it sees in the tree sends a message.
+- Replaces the old version of the tree with the new one in S3
+This is a sample of the message sent by the Lambda:
+```yaml
+{
+  "path": "Arabic/WMS_Arabic_1.xml",
+  "url": "https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/git/blobs/17f54c8c5ebccec643e23b54b3f4277f86204282",
+  "deleted": false 
+}
```

Comment:
Presumably you'd get a difference response if the file has been deleted?

Reply:
yes, will change, thanks